### DISCUSSION TITLE
Is it possible to generate html without printing to console

### DISCUSSION THREAD

**Discussion by sarnikowski** [Textualize/rich#1183] [Q&A] [answered]
First of all awesome tool !! :)

I am wondering whether it is possible to export to html without having to dump to the console at the same time. The documentation states that I have to set `record=True` for export to work, however in order to suppress the output I need to use `quiet=True`, which leaves my html export empty (as intended I guess). However trying to achieve this with for example a `Capture`, which I understand is the correct way to capture output without dumping it, I am stuck with `get()`, with no option of formatting. 

Is there a way to achieve this behavior, which I have overlooked? Thanks !

**Comment by willmcgugan** [comment, marked as answer]
Something along these lines should work:

```python
console = Console(record=True)
with console.capture():
    console.print("[b]Hello")

console.save_html("hello.html")
```

However, the most flexible way to create a console that exports HTML but never writes to the console would be to construct the Console with a file that goes nowhere. Something like this:

```python
import os
console = Console(record=True, file=open(os.devnull, "wt"), color_system="truecolor", width=100)
```

**Reply by sarnikowski** [reply]
I actually wrapped the block in `redirect_stdout` from contextlib, but this is much better. Thanks !
